### ISSUE TITLE
Add secondary table to hold Data Dictionary

### ISSUE THREAD

**Issue by karenmajewicz** [geobtaa/geoblacklight_admin#101]
This proposal is for a simple table to hold the Data Dictionary for datasets. Features:

- CSV import/export 
- Editable fields in GBL Admin
- Previewable in GeoBlacklight (as HTML probably)

**Comment by karenmajewicz** [comment]
I have not decided on the field names yet, but this is an example of a Data Dictionary
[btaa-6b8461b9-040c-4a7c-b09b-fde0c886c7f2_attribute-definitions.csv](https://github.com/user-attachments/files/17512456/btaa-6b8461b9-040c-4a7c-b09b-fde0c886c7f2_attribute-definitions.csv)

**Comment by karenmajewicz** [comment]
Here is another. I would make the column headers standardized before upload.[btaa-e776f983-39be-4417-81f4-9e91a6501c75.csv](https://github.com/user-attachments/files/17512464/btaa-e776f983-39be-4417-81f4-9e91a6501c75.csv)

**Comment by ewlarson** [comment]
Just a note to say let's lock down the initial fields/schema for this table. Should be relatively simple to implement.

**Comment by karenmajewicz** [comment]
Use these fields:

- label
- type
- values
- definition
- definition_source

These will all be strings.  Is it possible for the "values" field to be multivalued in some way?

**Comment by ewlarson** [comment]
Yes, the values field can be an array — PostgreSQL supports that well.

**Comment by karenmajewicz** [comment]
Also need friendlier_id in the table

**Comment by ewlarson** [comment]
Okay! Moving forward with these data dictionary attributes, unless you object:

* friendlier_id
* field_name — label or name here seem too abstract with "field_type"
* field_type — we can't use just "type" in Rails, as that's a reserved word
* values
* definition
* definition_source
* parent_id — for recursion, if necessary
